### PR TITLE
Chemical - Fix Virtual Zeus and units with damage turned off being killed by Chemical

### PR DESCRIPTION
**Pull request by YetheSamartaka**
**When merged this pull request will:**
- _Fix Virtual Zeus and units with damage turned off being killed by Chemical_
- _Add new function that clears chemical injuries that might be helpful in the future for modules and also other mods/scenarios_

### IMPORTANT

- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/chemical/functions/fnc_clearChemicalInjuriesLocal.sqf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+* [_unit] call kat_chemical_fnc_clearChemicalInjuriesLocal;
+*
+* Public: Yes
+*/
+
+params ["_unit"];
+
+_unit setVariable [QEGVAR(chemical,enteredPoison), false, true];
+_unit setVariable [QEGVAR(chemical,airPoisoning), false, true];
+_unit setVariable [QEGVAR(chemical,isTreated) ,true,true];
```
Comment (AtrixZockt) [inline addons/chemical/functions/fnc_clearChemicalInjuriesLocal.sqf]: spaces.  (Only Yethe will understand this: CTRL + SHIFT + K)

File: addons/chemical/functions/fnc_handleCSGas.sqf [hunk 2]
```diff
@@ -18,6 +18,9 @@ params ["_logic","_radius"];
 [{
     params["_params","_handler"];
     _params params["_unit"];
+    if (!isDamageAllowed _unit) exitWith {
+        [_unit] call FUNC(clearChemicalInjuriesLocal);    
+    };
```
Comment (AtrixZockt) [inline addons/chemical/functions/fnc_handleCSGas.sqf:24]: empty line after if statement to increase readability.

File: addons/chemical/functions/fnc_handleCSGas.sqf [hunk 3]
```diff
@@ -35,15 +38,18 @@ params ["_logic","_radius"];
...
-}, 5, [player]] call CBA_fnc_addPerFrameHandler;
+}, 5, [_unit]] call CBA_fnc_addPerFrameHandler;
 
 //Pos Check
 [{
     params ["_param","_handler"];
     _param params ["_logic","_radius","_unit"];
+    if (!isDamageAllowed _unit) exitWith {
+        [_unit] call FUNC(clearChemicalInjuriesLocal);
+    };
```
Comment (AtrixZockt) [inline addons/chemical/functions/fnc_handleCSGas.sqf:52]: same here

**Review by AtrixZockt — APPROVED** [review]
code style

File: addons/chemical/functions/fnc_handleCSGas.sqf (hunk 2 above)
Comment (YetheSamartaka) [reply addons/chemical/functions/fnc_handleCSGas.sqf:24]: addresed

**Pull request merged**